streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableGlobalKTableJoinTest.java

```diff
@@ -0,0 +1,156 @@
+        context = new NoOpProcessorContext();
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldNotForwardIfKeyNotInOtherStore() throws Exception {
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        processor.process("A", new Change<>("1", null));
+        final Change<String> a = (Change<String>) context.forwardedValues.get("A");
+        assertNull(a);
+    }
+
+    @Test
+    public void shouldJoinAndForwardIfKeyFromNewValueInOtherStore() throws Exception {
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        global.put("1", "B");
+        processor.process("A", new Change<>("1", null));
+        final Change<String> a = (Change<String>) context.forwardedValues.get("A");
+        assertEquals("1+B", a.newValue);
+        assertNull(a.oldValue);
+    }
+
+    @Test
+    public void shouldSendNewAndOldValuesCorrectly() throws Exception {
+        join.enableSendingOldValues();
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        global.put("1", "B");
+        global.put("2", "C");
+        processor.process("A", new Change<>("2", "1"));
+        final Change<String> a = (Change<String>) context.forwardedValues.get("A");
+        assertEquals("2+C", a.newValue);
+        assertEquals("1+B", a.oldValue);
+    }
+
+    @Test
+    public void shouldSendOldValueIfNewValueIsNullOldValueIsntNull() throws Exception {
+        join.enableSendingOldValues();
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        global.put("1", "B");
+        processor.process("A", new Change<>(null, "1"));
+        final Change<String> a = (Change<String>) context.forwardedValues.get("A");
+        assertNull(a.newValue);
+        assertEquals("1+B", a.oldValue);
+    }
+
+    @Test
+    public void shouldNotForwardIfDeleteAndOldKeyNotInOtherStoreAndSendOldValues() throws Exception {
+        join.enableSendingOldValues();
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        processor.process("A", new Change<>(null, "1"));
+        final Change<String> a = (Change<String>) context.forwardedValues.get("A");
+        assertNull(a);
+    }
+
+    @Test
+    public void shouldSendDeletesIfChangeHasOldValue() throws Exception {
+        global.put("1", "A");
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        processor.process("A", new Change<>(null, "1"));
+        final Change<String> a = (Change<String>) context.forwardedValues.get("A");
+        assertThat(a.newValue, is(nullValue()));
+        assertThat(a.oldValue, is(nullValue()));
+    }
+
+    @Test
+    public void shouldSendDeleteIfBothNewAndOldValuesAreNullButKeyMappingReturnsKey() throws Exception {
```

Comment:
Yep. I was thinking that a direct key mapping could result in the join, but that was incorrect.